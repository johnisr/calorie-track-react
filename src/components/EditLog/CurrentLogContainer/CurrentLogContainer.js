import React from 'react';
import { connect } from 'react-redux';
import { editCurrentEditLog, removeCurrentEditLog } from '../../../actions/currentLog';
import CurrentLogFoodListContainer from './CurrentLogFoodListContainer/CurrentLogFoodListContainer';
import { startEditLog, startRemoveLog } from '../../../actions/logs';
import { startEditFood } from '../../../actions/foods';
import CurrentLog from './CurrentLog/CurrentLog';

export class CurrentLogContainer extends React.Component {
  updateTimesUsed() {
    const foodUpdates = {};
    const oldLog = this.props.logs.filter(log => log.date === this.props.currentLog.date)[0];
    
    // Check if oldLog.foods exist because firebase doesn't store empty arrays
    if (oldLog.foods) {
      oldLog.foods.forEach((food) => {
        if (foodUpdates.hasOwnProperty(food.id)) {
          foodUpdates[food.id] = foodUpdates[food.id] - 1;
        } else {
          foodUpdates[food.id] = -1;
        }
      });
    };
    
    if (this.props.currentLog.foods) {
      this.props.currentLog.foods.forEach((food) => {
        if (foodUpdates.hasOwnProperty(food.id)) {
          foodUpdates[food.id] = foodUpdates[food.id] + 1;
        } else {
          foodUpdates[food.id] = 1;
        }
      });
    }

    if (this.props.foods) {
      this.props.foods.forEach((food) => {
        if (foodUpdates[food.id]) {
          this.props.startEditFood(
            food.id, 
            { timesUsed: food.timesUsed + foodUpdates[food.id] } 
          );
        }
      });
    }
  }

  isNumberWithAtMostTwoDecimals = (str) => {
    return str.match(/^\d{1,}?(\.\d{0,2})?$/);
  }
  handleWeightChange = (e) => {
    const weight = e.target.value;
    if (!weight || this.isNumberWithAtMostTwoDecimals(weight)) {
      this.props.editCurrentEditLog({ weight });
    }
  }
  handleLbsSelected = () => {
    const unit = 'lbs';
    this.props.editCurrentEditLog({ unit });
  }
  handleKgsSelected = () => {
    const unit = 'kgs';
    this.props.editCurrentEditLog({ unit });
  }
  handleSubmit = async () => {
    const weight = this.props.currentLog.weight !== '' ? parseFloat(this.props.currentLog.weight, 10) : 0;

    const foods = this.props.currentLog.foods;

    let total = { protein: 0, carbohydrates: 0, fat: 0, calories: 0 };
    if (foods) {
      total = foods.reduce((prev, curr) => ({
        protein: prev.protein + curr.protein,
        carbohydrates: prev.carbohydrates + curr.carbohydrates,
        fat: prev.fat + curr.fat,
        calories: prev.calories + curr.calories,
      }), total);
    }

    await this.props.editCurrentEditLog({ total });

    this.props.startEditLog(this.props.currentLog.date, {
      date: this.props.currentLog.date,
      unit: this.props.currentLog.unit,
      foods: foods ? foods : [],
      weight,
      total: {
        carbohydrates: +(total.carbohydrates).toFixed(2),
        protein: +(total.protein).toFixed(2),
        fat: +(total.fat).toFixed(2),
        calories: +(total.calories).toFixed(2),
      }
    });
    this.updateTimesUsed();
    
    this.props.removeCurrentEditLog();
    this.props.history.push('/Logs');
  }
  handleRemove = () => {
    this.props.startRemoveLog(this.props.currentLog.date);
    this.props.removeCurrentEditLog();
    this.props.history.push('/Logs');
  }
  render() {
    return (
      <div className="container">
        <div>
          <CurrentLogFoodListContainer />
        </div>
        <section className="section-center-6-start-end">
          <CurrentLog
            currentLog={this.props.currentLog}
            onWeightChange={this.handleWeightChange}
            onLbsSelected={this.handleLbsSelected}
            onKgsSelected={this.handleKgsSelected}
            onRemove={this.handleRemove}
            onSubmit={this.handleSubmit}
          />
        </section>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  currentLog: state.currentLog,
  logs: state.logs,
  foods: state.foods,
});

const mapDispatchToProps = (dispatch) => ({
  editCurrentEditLog: (updates) => dispatch(editCurrentEditLog(updates)),
  removeCurrentEditLog: () => dispatch(removeCurrentEditLog()),
  startEditLog: (date, log) => dispatch(startEditLog(date, log)),
  startEditFood: (id, addition) => dispatch(startEditFood(id, addition)),
  startRemoveLog: (date) => dispatch(startRemoveLog(date)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CurrentLogContainer);