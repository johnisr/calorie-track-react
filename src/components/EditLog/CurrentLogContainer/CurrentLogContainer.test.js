import React from 'react';
import { shallow } from 'enzyme';
import { CurrentLogContainer } from './CurrentLogContainer';
import logsData from '../../../tests/fixtures/logs';
import foodsData from '../../../tests/fixtures/foods';

let editCurrentEditLog,
removeCurrentEditLog,
startEditLog,
startEditFood,
startRemoveLog,
currentLog,
logs,
foods,
history,
wrapper;

beforeEach(() => {
  editCurrentEditLog = jest.fn();
  removeCurrentEditLog = jest.fn();
  startEditLog = jest.fn();
  startEditFood = jest.fn();
  startRemoveLog = jest.fn();
  currentLog = logsData[2];
  logs = logsData;
  foods = foodsData;
  history = { push: jest.fn() };
  wrapper = shallow(<CurrentLogContainer
    editCurrentEditLog={editCurrentEditLog}
    currentLog={currentLog}
    removeCurrentEditLog={removeCurrentEditLog}
    startEditLog={startEditLog}
    startEditFood={startEditFood}
    startRemoveLog={startRemoveLog}
    history={history}
    logs={logs}
    foods={foods}
  />)
});

it('should render CurrentLogContainer', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handleRemove log', () => {
  wrapper.find('currentLog').prop('onRemove')();
  expect(startRemoveLog).toHaveBeenLastCalledWith(logsData[2].date);
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handleSubmit with empty foods', async () => {
  currentLog = {
    date: 0,
    weight: 0,
    unit: 'lb',
    foods: [],
    total: {
        protein: 0,
        carbohydrates: 0,
        fat: 0,
        calories: 0,
    }
  };
  wrapper = shallow(<CurrentLogContainer
    editCurrentEditLog={editCurrentEditLog}
    removeCurrentEditLog={removeCurrentEditLog}
    startEditLog={startEditLog}
    startEditFood={startEditFood}
    startRemoveLog={startRemoveLog}
    history={history}
    logs={logsData}
    foods={foods}
    currentLog={currentLog}
  />)

  await wrapper.find('currentLog').prop('onSubmit')();
  expect.assertions(4);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ total: currentLog.total });
  expect(startEditLog).toHaveBeenLastCalledWith(currentLog.date, currentLog);
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handleSubmit with foods and no change', async () => {
  await wrapper.find('currentLog').prop('onSubmit')();
  expect.assertions(5);
 expect(editCurrentEditLog).toHaveBeenLastCalledWith({ total: currentLog.total });
  expect(startEditLog).toHaveBeenLastCalledWith(currentLog.date, currentLog);
  expect(startEditFood).not.toHaveBeenCalled();
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handleSubmit with a change to other fields', async () => {
  currentLog = {
    ...logsData[2],
    weight: 300,
    unit: 'kg',
  };
  
  wrapper = shallow(<CurrentLogContainer
    editCurrentEditLog={editCurrentEditLog}
    removeCurrentEditLog={removeCurrentEditLog}
    startEditLog={startEditLog}
    startEditFood={startEditFood}
    startRemoveLog={startRemoveLog}
    history={history}
    logs={logsData}
    foods={foods}
    currentLog={currentLog}
  />)

  await wrapper.find('currentLog').prop('onSubmit')();
  expect.assertions(4);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ total: currentLog.total });
  expect(startEditLog).toHaveBeenLastCalledWith(currentLog.date, currentLog);
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handleSubmit with foods and a change (removing food)', async () => {
  currentLog = {
    ...logsData[2],
    foods: [
      foodsData[1],
    ],
  };
  const total = {
    carbohydrates: 0,
    protein: 31,
    fat: 3.6,
    calories: 165,
  };
  wrapper = shallow(<CurrentLogContainer
    editCurrentEditLog={editCurrentEditLog}
    removeCurrentEditLog={removeCurrentEditLog}
    startEditLog={startEditLog}
    startEditFood={startEditFood}
    startRemoveLog={startRemoveLog}
    history={history}
    logs={logsData}
    foods={foods}
    currentLog={currentLog}
  />)

  await wrapper.find('currentLog').prop('onSubmit')();
  expect.assertions(5);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ total });
  expect(startEditLog).toHaveBeenLastCalledWith(currentLog.date, {
    date: currentLog.date,
    foods: currentLog.foods,
    total,
    unit: currentLog.unit,
    weight: currentLog.weight,
  });
  // food with id 3 (almonds) went from 2 timesUsed to 1 because it was left out
  expect(startEditFood).toHaveBeenLastCalledWith('3', {timesUsed: 1 });
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handleSubmit with foods and a change (adding food)', async () => {
  currentLog = {
    ...logsData[2],
    foods: [
      foods[0],
      foods[1],
      foods[2],
      foods[2],
    ],
  };
  const total = {
    carbohydrates: 49,
    protein: 47.2,
    fat: 32,
    calories: 696,
  };
  wrapper = shallow(<CurrentLogContainer
    editCurrentEditLog={editCurrentEditLog}
    removeCurrentEditLog={removeCurrentEditLog}
    startEditLog={startEditLog}
    startEditFood={startEditFood}
    startRemoveLog={startRemoveLog}
    history={history}
    logs={logsData}
    foods={foods}
    currentLog={currentLog}
  />)

  await wrapper.find('currentLog').prop('onSubmit')();
  expect.assertions(5);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ total });
  expect(startEditLog).toHaveBeenLastCalledWith(currentLog.date, {
    date: currentLog.date,
    foods: currentLog.foods,
    total,
    unit: currentLog.unit,
    weight: currentLog.weight,
  });
  // food with id 3 (almonds) went from 2 timesUsed to 3 because another one added
  expect(startEditFood).toHaveBeenLastCalledWith('3', {timesUsed: 3 });
  expect(removeCurrentEditLog).toHaveBeenCalled();
  expect(history.push).toHaveBeenLastCalledWith('/Logs');
});

it('should handle Kgs Selected', () => {
  wrapper.find('currentLog').prop('onKgsSelected')();
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ unit: 'kgs' });
});

it('should handle Lbs Selected', () => {
  wrapper.find('currentLog').prop('onLbsSelected')();
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ unit: 'lbs' });
});

it('should handle Weight change with valid weight (integer)', () => {
  const e = {target: { value: '11' }};
  wrapper.find('currentLog').prop('onWeightChange')(e);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ weight: e.target.value });
});

it('should handle Weight change with valid weight (float, 2 decimals)', () => {
  const e = {target: { value: '200.11' }};
  wrapper.find('currentLog').prop('onWeightChange')(e);
  expect(editCurrentEditLog).toHaveBeenLastCalledWith({ weight: e.target.value });
});

it('should not handle Weight change with invalid weight', () => {
  const e = {target: { value: 'abc' }};
  wrapper.find('currentLog').prop('onWeightChange')(e);
  expect(editCurrentEditLog).not.toHaveBeenCalled();
});

it('should not handle Weight change with invalid weight (float, 3 decimals)', () => {
  const e = {target: { value: '200.123' }};
  wrapper.find('currentLog').prop('onWeightChange')(e);
  expect(editCurrentEditLog).not.toHaveBeenCalled();
});