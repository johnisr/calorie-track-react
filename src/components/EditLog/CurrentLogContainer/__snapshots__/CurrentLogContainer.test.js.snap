// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentLogContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentLogContainer
    currentLog={
        Object {
            "date": 86400000,
            "foods": Array [
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
              Object {
                "amount": 100,
                "base": Object {
                  "amount": 100,
                  "calories": 165,
                  "carbohydrates": 0,
                  "fat": 3.6,
                  "protein": 31,
                },
                "calories": 165,
                "carbohydrates": 0,
                "createdAt": 1000,
                "fat": 3.6,
                "id": "2",
                "name": "Chicken Breast",
                "protein": 31,
                "timesUsed": 1,
                "unit": "grams",
              },
              Object {
                "amount": 24,
                "calories": 163,
                "carbohydrates": 2.5,
                "createdAt": -1000,
                "fat": 14,
                "id": "3",
                "name": "Almonds",
                "protein": 6,
                "timesUsed": 2,
                "unit": "pieces",
              },
            ],
            "total": Object {
              "calories": 533,
              "carbohydrates": 46.5,
              "fat": 18,
              "protein": 41.2,
            },
            "unit": "lb",
            "weight": 201,
          }
    }
    editCurrentEditLog={[Function]}
    foods={
        Array [
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
            Object {
              "amount": 100,
              "base": Object {
                "amount": 100,
                "calories": 165,
                "carbohydrates": 0,
                "fat": 3.6,
                "protein": 31,
              },
              "calories": 165,
              "carbohydrates": 0,
              "createdAt": 1000,
              "fat": 3.6,
              "id": "2",
              "name": "Chicken Breast",
              "protein": 31,
              "timesUsed": 1,
              "unit": "grams",
            },
            Object {
              "amount": 24,
              "calories": 163,
              "carbohydrates": 2.5,
              "createdAt": -1000,
              "fat": 14,
              "id": "3",
              "name": "Almonds",
              "protein": 6,
              "timesUsed": 2,
              "unit": "pieces",
            },
          ]
    }
    history={
        Object {
            "push": [Function],
          }
    }
    logs={
        Array [
            Object {
              "date": -154800000,
              "foods": Array [
                Object {
                  "amount": 24,
                  "calories": 163,
                  "carbohydrates": 2.5,
                  "createdAt": -1000,
                  "fat": 14,
                  "id": "3",
                  "name": "Almonds",
                  "protein": 6,
                  "timesUsed": 2,
                  "unit": "pieces",
                },
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 100,
                  "base": Object {
                    "amount": 100,
                    "calories": 165,
                    "carbohydrates": 0,
                    "fat": 3.6,
                    "protein": 31,
                  },
                  "calories": 165,
                  "carbohydrates": 0,
                  "createdAt": 1000,
                  "fat": 3.6,
                  "id": "2",
                  "name": "Chicken Breast",
                  "protein": 31,
                  "timesUsed": 1,
                  "unit": "grams",
                },
              ],
              "total": Object {
                "calories": 533,
                "carbohydrates": 46.5,
                "fat": 18,
                "protein": 41.2,
              },
              "unit": "lb",
              "weight": 202,
            },
            Object {
              "date": 0,
              "foods": Array [
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
              ],
              "total": Object {
                "calories": 410,
                "carbohydrates": 88,
                "fat": 0.8,
                "protein": 8.4,
              },
              "unit": "lb",
              "weight": 200,
            },
            Object {
              "date": 86400000,
              "foods": Array [
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 100,
                  "base": Object {
                    "amount": 100,
                    "calories": 165,
                    "carbohydrates": 0,
                    "fat": 3.6,
                    "protein": 31,
                  },
                  "calories": 165,
                  "carbohydrates": 0,
                  "createdAt": 1000,
                  "fat": 3.6,
                  "id": "2",
                  "name": "Chicken Breast",
                  "protein": 31,
                  "timesUsed": 1,
                  "unit": "grams",
                },
                Object {
                  "amount": 24,
                  "calories": 163,
                  "carbohydrates": 2.5,
                  "createdAt": -1000,
                  "fat": 14,
                  "id": "3",
                  "name": "Almonds",
                  "protein": 6,
                  "timesUsed": 2,
                  "unit": "pieces",
                },
              ],
              "total": Object {
                "calories": 533,
                "carbohydrates": 46.5,
                "fat": 18,
                "protein": 41.2,
              },
              "unit": "lb",
              "weight": 201,
            },
          ]
    }
    removeCurrentEditLog={[Function]}
    startEditFood={[Function]}
    startEditLog={[Function]}
    startRemoveLog={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Connect(CurrentLogFoodListContainer) />
</div>,
        <section
          className="section-center-6-start-end"
>
          <currentLog
                    currentLog={
                              Object {
                                        "date": 86400000,
                                        "foods": Array [
                                          Object {
                                            "amount": 200,
                                            "calories": 205,
                                            "carbohydrates": 44,
                                            "createdAt": 0,
                                            "fat": 0.4,
                                            "id": "1",
                                            "name": "Rice (Jasmine)",
                                            "protein": 4.2,
                                            "timesUsed": 0,
                                            "unit": "grams",
                                          },
                                          Object {
                                            "amount": 100,
                                            "base": Object {
                                              "amount": 100,
                                              "calories": 165,
                                              "carbohydrates": 0,
                                              "fat": 3.6,
                                              "protein": 31,
                                            },
                                            "calories": 165,
                                            "carbohydrates": 0,
                                            "createdAt": 1000,
                                            "fat": 3.6,
                                            "id": "2",
                                            "name": "Chicken Breast",
                                            "protein": 31,
                                            "timesUsed": 1,
                                            "unit": "grams",
                                          },
                                          Object {
                                            "amount": 24,
                                            "calories": 163,
                                            "carbohydrates": 2.5,
                                            "createdAt": -1000,
                                            "fat": 14,
                                            "id": "3",
                                            "name": "Almonds",
                                            "protein": 6,
                                            "timesUsed": 2,
                                            "unit": "pieces",
                                          },
                                        ],
                                        "total": Object {
                                          "calories": 533,
                                          "carbohydrates": 46.5,
                                          "fat": 18,
                                          "protein": 41.2,
                                        },
                                        "unit": "lb",
                                        "weight": 201,
                                      }
                    }
                    onKgsSelected={[Function]}
                    onLbsSelected={[Function]}
                    onRemove={[Function]}
                    onSubmit={[Function]}
                    onWeightChange={[Function]}
          />
</section>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(CurrentLogFoodListContainer) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <currentLog
            currentLog={
                        Object {
                                    "date": 86400000,
                                    "foods": Array [
                                      Object {
                                        "amount": 200,
                                        "calories": 205,
                                        "carbohydrates": 44,
                                        "createdAt": 0,
                                        "fat": 0.4,
                                        "id": "1",
                                        "name": "Rice (Jasmine)",
                                        "protein": 4.2,
                                        "timesUsed": 0,
                                        "unit": "grams",
                                      },
                                      Object {
                                        "amount": 100,
                                        "base": Object {
                                          "amount": 100,
                                          "calories": 165,
                                          "carbohydrates": 0,
                                          "fat": 3.6,
                                          "protein": 31,
                                        },
                                        "calories": 165,
                                        "carbohydrates": 0,
                                        "createdAt": 1000,
                                        "fat": 3.6,
                                        "id": "2",
                                        "name": "Chicken Breast",
                                        "protein": 31,
                                        "timesUsed": 1,
                                        "unit": "grams",
                                      },
                                      Object {
                                        "amount": 24,
                                        "calories": 163,
                                        "carbohydrates": 2.5,
                                        "createdAt": -1000,
                                        "fat": 14,
                                        "id": "3",
                                        "name": "Almonds",
                                        "protein": 6,
                                        "timesUsed": 2,
                                        "unit": "pieces",
                                      },
                                    ],
                                    "total": Object {
                                      "calories": 533,
                                      "carbohydrates": 46.5,
                                      "fat": 18,
                                      "protein": 41.2,
                                    },
                                    "unit": "lb",
                                    "weight": 201,
                                  }
            }
            onKgsSelected={[Function]}
            onLbsSelected={[Function]}
            onRemove={[Function]}
            onSubmit={[Function]}
            onWeightChange={[Function]}
/>,
          "className": "section-center-6-start-end",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentLog": Object {
              "date": 86400000,
              "foods": Array [
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 100,
                  "base": Object {
                    "amount": 100,
                    "calories": 165,
                    "carbohydrates": 0,
                    "fat": 3.6,
                    "protein": 31,
                  },
                  "calories": 165,
                  "carbohydrates": 0,
                  "createdAt": 1000,
                  "fat": 3.6,
                  "id": "2",
                  "name": "Chicken Breast",
                  "protein": 31,
                  "timesUsed": 1,
                  "unit": "grams",
                },
                Object {
                  "amount": 24,
                  "calories": 163,
                  "carbohydrates": 2.5,
                  "createdAt": -1000,
                  "fat": 14,
                  "id": "3",
                  "name": "Almonds",
                  "protein": 6,
                  "timesUsed": 2,
                  "unit": "pieces",
                },
              ],
              "total": Object {
                "calories": 533,
                "carbohydrates": 46.5,
                "fat": 18,
                "protein": 41.2,
              },
              "unit": "lb",
              "weight": 201,
            },
            "onKgsSelected": [Function],
            "onLbsSelected": [Function],
            "onRemove": [Function],
            "onSubmit": [Function],
            "onWeightChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Connect(CurrentLogFoodListContainer) />
</div>,
          <section
            className="section-center-6-start-end"
>
            <currentLog
                        currentLog={
                                    Object {
                                                "date": 86400000,
                                                "foods": Array [
                                                  Object {
                                                    "amount": 200,
                                                    "calories": 205,
                                                    "carbohydrates": 44,
                                                    "createdAt": 0,
                                                    "fat": 0.4,
                                                    "id": "1",
                                                    "name": "Rice (Jasmine)",
                                                    "protein": 4.2,
                                                    "timesUsed": 0,
                                                    "unit": "grams",
                                                  },
                                                  Object {
                                                    "amount": 100,
                                                    "base": Object {
                                                      "amount": 100,
                                                      "calories": 165,
                                                      "carbohydrates": 0,
                                                      "fat": 3.6,
                                                      "protein": 31,
                                                    },
                                                    "calories": 165,
                                                    "carbohydrates": 0,
                                                    "createdAt": 1000,
                                                    "fat": 3.6,
                                                    "id": "2",
                                                    "name": "Chicken Breast",
                                                    "protein": 31,
                                                    "timesUsed": 1,
                                                    "unit": "grams",
                                                  },
                                                  Object {
                                                    "amount": 24,
                                                    "calories": 163,
                                                    "carbohydrates": 2.5,
                                                    "createdAt": -1000,
                                                    "fat": 14,
                                                    "id": "3",
                                                    "name": "Almonds",
                                                    "protein": 6,
                                                    "timesUsed": 2,
                                                    "unit": "pieces",
                                                  },
                                                ],
                                                "total": Object {
                                                  "calories": 533,
                                                  "carbohydrates": 46.5,
                                                  "fat": 18,
                                                  "protein": 41.2,
                                                },
                                                "unit": "lb",
                                                "weight": 201,
                                              }
                        }
                        onKgsSelected={[Function]}
                        onLbsSelected={[Function]}
                        onRemove={[Function]}
                        onSubmit={[Function]}
                        onWeightChange={[Function]}
            />
</section>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(CurrentLogFoodListContainer) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <currentLog
              currentLog={
                            Object {
                                          "date": 86400000,
                                          "foods": Array [
                                            Object {
                                              "amount": 200,
                                              "calories": 205,
                                              "carbohydrates": 44,
                                              "createdAt": 0,
                                              "fat": 0.4,
                                              "id": "1",
                                              "name": "Rice (Jasmine)",
                                              "protein": 4.2,
                                              "timesUsed": 0,
                                              "unit": "grams",
                                            },
                                            Object {
                                              "amount": 100,
                                              "base": Object {
                                                "amount": 100,
                                                "calories": 165,
                                                "carbohydrates": 0,
                                                "fat": 3.6,
                                                "protein": 31,
                                              },
                                              "calories": 165,
                                              "carbohydrates": 0,
                                              "createdAt": 1000,
                                              "fat": 3.6,
                                              "id": "2",
                                              "name": "Chicken Breast",
                                              "protein": 31,
                                              "timesUsed": 1,
                                              "unit": "grams",
                                            },
                                            Object {
                                              "amount": 24,
                                              "calories": 163,
                                              "carbohydrates": 2.5,
                                              "createdAt": -1000,
                                              "fat": 14,
                                              "id": "3",
                                              "name": "Almonds",
                                              "protein": 6,
                                              "timesUsed": 2,
                                              "unit": "pieces",
                                            },
                                          ],
                                          "total": Object {
                                            "calories": 533,
                                            "carbohydrates": 46.5,
                                            "fat": 18,
                                            "protein": 41.2,
                                          },
                                          "unit": "lb",
                                          "weight": 201,
                                        }
              }
              onKgsSelected={[Function]}
              onLbsSelected={[Function]}
              onRemove={[Function]}
              onSubmit={[Function]}
              onWeightChange={[Function]}
/>,
            "className": "section-center-6-start-end",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentLog": Object {
                "date": 86400000,
                "foods": Array [
                  Object {
                    "amount": 200,
                    "calories": 205,
                    "carbohydrates": 44,
                    "createdAt": 0,
                    "fat": 0.4,
                    "id": "1",
                    "name": "Rice (Jasmine)",
                    "protein": 4.2,
                    "timesUsed": 0,
                    "unit": "grams",
                  },
                  Object {
                    "amount": 100,
                    "base": Object {
                      "amount": 100,
                      "calories": 165,
                      "carbohydrates": 0,
                      "fat": 3.6,
                      "protein": 31,
                    },
                    "calories": 165,
                    "carbohydrates": 0,
                    "createdAt": 1000,
                    "fat": 3.6,
                    "id": "2",
                    "name": "Chicken Breast",
                    "protein": 31,
                    "timesUsed": 1,
                    "unit": "grams",
                  },
                  Object {
                    "amount": 24,
                    "calories": 163,
                    "carbohydrates": 2.5,
                    "createdAt": -1000,
                    "fat": 14,
                    "id": "3",
                    "name": "Almonds",
                    "protein": 6,
                    "timesUsed": 2,
                    "unit": "pieces",
                  },
                ],
                "total": Object {
                  "calories": 533,
                  "carbohydrates": 46.5,
                  "fat": 18,
                  "protein": 41.2,
                },
                "unit": "lb",
                "weight": 201,
              },
              "onKgsSelected": [Function],
              "onLbsSelected": [Function],
              "onRemove": [Function],
              "onSubmit": [Function],
              "onWeightChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
